pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -830,17 +831,22 @@ func isIncomplete(condition *v2vv1.VirtualMachineImportCondition) bool {
 }
 
 func (r *ReconcileVirtualMachineImport) createProvider(vmi *v2vv1.VirtualMachineImport) (provider.Provider, error) {
+	config, err := r.ctrlConfigProvider.GetConfig()
+	if err != nil {
+		log.Error(err, "Cannot get controller config.")
+	}
+
 	// The type of the provider is evaluated based on the source field from the CR
 	if vmi.Spec.Source.Ovirt != nil {
-		config, err := r.ctrlConfigProvider.GetConfig()
-		if err != nil {
-			log.Error(err, "Cannot get controller config.")
-		}
 		provider := ovirtprovider.NewOvirtProvider(vmi.ObjectMeta, vmi.TypeMeta, r.client, r.ocClient, r.factory, r.kvConfigProvider, config)
 		return &provider, nil
 	}
+	if vmi.Spec.Source.Vmware != nil {
```

Comment:
There is a small chance that both ovirt and vmware might be provided (if it isn't enforced by the schema validation - could be validated by 'oneOf'). In that case we should deny the request.